src/app/interfaces/AdvancedConfig.ts

```diff
@@ -0,0 +1,29 @@
+export interface AdvancedConfig {
+    advancedmode?: boolean;
+    anonstats?: boolean;
+    anonstats_token?: string;
+    autotune?: boolean;
+    boot_scrub?: number;
+    consolemenu?: boolean;
+    consolemsg?: boolean;
+    debugkernel?: boolean;
+    fqdn_syslog?: boolean;
+    id?: number;
+    isolated_gpu_pci_ids?: string[];
+    kdump_enabled?: boolean;
+    motd?: string;
+    overprovision?: number;
+    powerdaemon?: boolean;
+    sed_user?: string;
+    serialconsole?: boolean;
+    serialport?: string;
+    serialspeed?: string;
+    swapondrive?: number;
+    syslog_tls_certificate?: number;
+    syslog_transport?: string;
+    sysloglevel?: string;
+    syslogserver?: string;
+    traceback?: boolean;
+    uploadcrash?: boolean;
+    sed_passwd?: string;
```

Comment:
Is everything really optional?

Reply:
Everything has `required: false` in `api/docs` which is why I kept it optional. Not sure if that means it won't be in the response or it will be null. For reference, there are fields with `type` `number` or `null`. Those might be nullable. The rest might not exist. At least that's my understanding.

Reply:
An example is `overprovision` for `system.advanced.update` middleware. According to `api/docs`

`
"overprovision": {
   "type": [
    "integer",
    "null"
   ],
   "_name_": "overprovision",
   "title": "overprovision",
   "_required_": false
  }
`

Reply:
Right, for update methods most of the times the attributes are optional, meaning you only need to submit the attribute you're changing

Reply:
Okay, sounds good.

Reply:
The attributes are returned at all times

Reply:
@undsoft Ideally, we'd like to use the same interface for both `get` and `update` calls. It might make sense to adhere to the most relaxed definition that we can get away with.

Reply:
I hear ya, but there's another thing we can do.
We can define this interface properly without question marks, and then add:
```
export type AdvancedConfigUpdate = Partial<AdvancedConfig>;
```

This way we can be sure where we to expect fields or not.

More info: https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
> The attributes are returned at all times
@undsoft let me know if I should remove the optional feature of those attributes